tests/models/test_pure_svd.py

```diff
@@ -194,8 +194,10 @@ def test_i2i(
             actual,
         )
 
-    def test_second_fit_refits_model(self, dataset: Dataset) -> None:
-        model = PureSVDModel(factors=3)
+    @pytest.mark.parametrize("tol, random_state", ((0, None), (0, 42), (0.3, 42)))
+    def test_second_fit_refits_model(self, dataset: Dataset, tol: float, random_state: tp.Optional[int]) -> None:
+        # If `tol` is greater than 0, and `random_state` is None, the results can be different.
```

Comment:
What actually happens in this case?

Reply:
You mean why the results are different?

Reply:
What do you mean by different results? Second `fit` doesn't refit model? Or we just can't check it because random state is not set and we get different results each fit.

Reply:
If random state is not set we get different results each fit (it's not directly connected with refit).  It worked before because results are the same even without random state if `tol == 0`